Tensile/Tests/unit/test_TensileCreateLibrary.py

```diff
@@ -175,3 +175,103 @@ def test_verifyManifest():
   
     with open("bar.asm", "x"):
       assert TensileCreateLibrary.verifyManifest(manifestFile), "files in manifest should be on disk"
+
+def test_parseGfxArchitectures():
+    # Test `all` architectures
+    archs = 'all'
+    result = TensileCreateLibrary.parseGfxArchitectures(archs)
+    expected = set([archs])
+    assert result == expected, f"arch `{archs}` should parse to {expected} but instead maps to {result}"
+
+    expected = set([item for item in filter(lambda x: x != "all", Common.architectureMap.keys())])
+    # Test all architectures, but supplied separately with semicolon delimieter (cli usage)
+    archs = "gfx000;gfx803;gfx900;gfx900:xnack-;gfx906;gfx906:xnack+;gfx906:xnack-;gfx908;gfx908:xnack+;"+\
+            "gfx908:xnack-;gfx90a;gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx940:xnack+;gfx940:xnack-;gfx941;"+\
+            "gfx941:xnack+;gfx941:xnack-;gfx942;gfx942:xnack+;gfx942:xnack-;gfx1010;gfx1011;gfx1012;"+\
+            "gfx1030;gfx1031;gfx1032;gfx1034;gfx1035;gfx1100;gfx1101;gfx1102"
+    result = TensileCreateLibrary.parseGfxArchitectures(archs)
+    assert result == expected, f"arch `{archs}` should map to {expected} but instead maps to {result}"
```

Comment:
Do we want a test assertation that checks if the return value is a set?

Reply:
By uniqueness you mean enforced one-to-one mapping between each Gfx name and common name?